src/main/java/org/folio/circulation/domain/Account.java

```diff
@@ -210,27 +224,27 @@ public String getTransferAccountName() {
 
   public Account subtractRemainingAmount(FeeAmount toSubtract) {
     return new Account(this.id, relatedRecordsInfo, amount, remaining.subtract(toSubtract),
-      status, paymentStatus, feeFineActions);
+      status, paymentStatus, feeFineActions, createdAt);
   }
 
   public Account addRemainingAmount(FeeAmount toAdd) {
     return new Account(this.id, relatedRecordsInfo, amount, remaining.add(toAdd),
-      status, paymentStatus, feeFineActions);
+      status, paymentStatus, feeFineActions, createdAt);
   }
 
   private Account withStatus(AccountStatus status) {
     return new Account(id, relatedRecordsInfo, amount, remaining, status.getValue(),
-      paymentStatus, feeFineActions);
+      paymentStatus, feeFineActions, createdAt);
```

Comment:
Just a FYI: We've introduced `lombok` dependency. It can be now replaced with `@With` annotation.

Reply:
Yes, I saw that. A very welcome change.